snowfakery/data_generator_runtime_object_model.py

```diff
@@ -294,6 +295,7 @@ def __init__(self, function_name, args, filename, line_num):
             self.kwargs = {}
 
     def render(self, context: RuntimeContext) -> FieldValue:
+        context.unique_context_identifier = id(self)
```

Comment:
Are you sure that contexts are never garbage collected during the lifetime of when you are using this identifier? If they are, the id might be reused.